lib/valid_email2.rb

```diff
@@ -8,16 +8,24 @@ module ValidEmail2
   DISPOSABLE_FILE = File.expand_path('../config/disposable_email_domains.txt', __dir__)
 
   class << self
+    attr_accessor :disposable_proc, :deny_proc, :allow_proc
+
+    def reset_lists
+      @disposable_emails = nil
+      @deny_list = nil
+      @allow_list = nil
+    end
+
     def disposable_emails
-      @disposable_emails ||= load_file(DISPOSABLE_FILE)
+      @disposable_emails ||= disposable_proc&.call || load_file(DISPOSABLE_FILE)
```

Comment:
I think we should be cautious of memoizing the lists if it's provided via a proc.
As a developer i would expect the proc to be called each time and that it would be up to me to cache or memoize the list myself.

Reply:
Changed it accordingly